File: titanium-rdf-api/src/main/scala/eu/ostrzyciel/jelly/convert/titanium/internal/TitaniumConverterFactory.scala
package eu.ostrzyciel.jelly.convert.titanium.internal

import eu.ostrzyciel.jelly.convert.titanium.internal.TitaniumRdf.*
import eu.ostrzyciel.jelly.core.ConverterFactory

private[titanium] object TitaniumConverterFactory
  extends ConverterFactory[TitaniumEncoderConverter, TitaniumDecoderConverter, Node, String, Triple, Quad]:

  /**
   * @inheritdoc
   */
  override lazy val decoderConverter: TitaniumDecoderConverter = new TitaniumDecoderConverter()

  /**
   * @inheritdoc
   */
  override lazy val encoderConverter: TitaniumEncoderConverter = new TitaniumEncoderConverter()

File: titanium-rdf-api/src/main/java/eu/neverblink/jelly/convert/titanium/internal/TitaniumConverterFactory.java
package eu.neverblink.jelly.convert.titanium.internal;

import eu.neverblink.jelly.core.InternalApi;
import eu.neverblink.jelly.core.JellyConverterFactory;

/**
 * A singleton factory for creating Jelly Titanium converters.
 * <p>
 * This class is a singleton and should be accessed via the {@link #getInstance()} method.
 */
@InternalApi
public final class TitaniumConverterFactory
    extends JellyConverterFactory<Object, String, TitaniumEncoderConverter, TitaniumDecoderConverter> {

    private static final TitaniumConverterFactory INSTANCE = new TitaniumConverterFactory();

    private TitaniumConverterFactory() {}

    /**
     * Returns the singleton instance of the {@link TitaniumConverterFactory}.
     *
     * @return the singleton instance of the {@link TitaniumConverterFactory}
     */
    public static TitaniumConverterFactory getInstance() {
        return INSTANCE;
    }

    @Override
    public TitaniumEncoderConverter encoderConverter() {
        return new TitaniumEncoderConverter();
    }

    @Override
    public TitaniumDecoderConverter decoderConverter() {
        return new TitaniumDecoderConverter();
    }
}
